staging/src/k8s.io/cloud-provider/plugins.go

```diff
@@ -87,8 +87,20 @@ func GetCloudProvider(name string, config io.Reader) (Interface, error) {
 }
 
 // Detects if the string is an external cloud provider
-func IsExternal(name string) bool {
-	return name == externalCloudProvider
+// IsExternal is responsible for serving DisableCloudProviders featureGate and stop cloud.Interface
+// from being initialized in kubelet, kube-controller-manager or kube-api-server
+func IsExternal(name string, disableCloudProvidersSet bool) bool {
+	switch name {
+	case "aws", "azure", "gce", "openstack", "vsphere":
+		if disableCloudProvidersSet {
+			klog.Fatalf("Cloud provider %q was specified, but in-tree cloud providers are disabled. Please move to --cloud-provider=external and out-of-tree CCM/CSI in your cluster.", name)
```

Comment:
Rather than fataling in library code, let's update the method signature to return an error.  There are relatively few callers.